Dead links

Using pytest-check-links:

```
C:\dev\jupyterlab [master ≡]> py.test --check-links -k .md .
============================= test session starts =============================
platform win32 -- Python 3.5.2, pytest-3.0.6, py-1.4.32, pluggy-0.4.0
rootdir: C:\dev\jupyterlab, inifile: pytest.ini
plugins: tornado-0.4.5, timeout-1.2.0, cov-2.4.0, nbval-0.4.1, check-links-0.0.3.dev4
collected 121 items

CONTRIBUTING.md .......F..
README.md ......................
RELEASE.md ..
design\about.md .
tutorial\adding_content.md ...F
tutorial\documents.md F.........
tutorial\examples.md ..
tutorial\labextensions.md F.
tutorial\notebook.md ..................
tutorial\patterns.md ..
tutorial\plugins.md ...............
tutorial\repo.md .
tutorial\virtualdom.md ..

================================== FAILURES ===================================
___ C:\dev\jupyterlab\CONTRIBUTING.md: http://jupyter.github.io/jupyterlab/ ___
http://jupyter.github.io/jupyterlab/: HTTP Error 404: Not Found
________ C:\dev\jupyterlab\tutorial\adding_content.md: documents.html _________
documents.html: No such file: C:\dev\jupyterlab\tutorial\documents.html
____________ C:\dev\jupyterlab\tutorial\documents.md: plugins.html ____________
plugins.html: No such file: C:\dev\jupyterlab\tutorial\plugins.html
__________ C:\dev\jupyterlab\tutorial\labextensions.md: plugins.html __________
plugins.html: No such file: C:\dev\jupyterlab\tutorial\plugins.html
============================= 30 tests deselected =============================
============= 4 failed, 87 passed, 30 deselected in 40.90 seconds =============
```
The last three are probably ok(?), but the first one is properly broken.

--- a/scripts/travis_script.sh
+++ b/scripts/travis_script.sh
@@ -40,7 +40,7 @@ if [[ $GROUP == coverage_and_docs ]]; then
 
     # Run the link check
     pip install -q pytest-check-links
-    py.test --check-links -k .md .
+    py.test --check-links **/*.md
 
     # Build the docs
     npm run docs

--- a/scripts/travis_script.sh
+++ b/scripts/travis_script.sh
@@ -40,7 +40,7 @@ if [[ $GROUP == coverage_and_docs ]]; then
 
     # Run the link check
     pip install -q pytest-check-links
-    py.test --check-links **/*.md
+    py.test --check-links -k .md
 
     # Build the docs
     npm run docs

--- a/scripts/travis_script.sh
+++ b/scripts/travis_script.sh
@@ -40,7 +40,7 @@ if [[ $GROUP == coverage_and_docs ]]; then
 
     # Run the link check
     pip install -q pytest-check-links
-    py.test --check-links -k .md
+    py.test --check-links -k .md .
 
     # Build the docs
     npm run docs

--- a/scripts/travis_script.sh
+++ b/scripts/travis_script.sh
@@ -38,6 +38,10 @@ if [[ $GROUP == coverage_and_docs ]]; then
     # Run the coverage check.
     npm run test:coverage
 
+    # Run the link check
+    pip install -q pytest-check-links
+    py.test --check-links -k .md .
+
     # Build the docs
     npm run docs
     cp jupyter-plugins-demo.gif docs

--- a/tutorial/adding_content.md
+++ b/tutorial/adding_content.md
@@ -15,7 +15,7 @@ files: `npm install -g tsd`.
 - If creating CSS, import them in `src/default-themes/index.css`.
 - The `index.ts` file should have the core logic for the plugin.  In
 this case, it should create a widget and widget factory for rendering geojson 
-files (see [Documents](documents.html)).
+files (see [Documents](documents.md)).
 - The `plugin.ts` file should create the extension and add the content
 to the application.  In this case registering the widget factory using
 the document registry.

--- a/tutorial/conf.py
+++ b/tutorial/conf.py
@@ -23,6 +23,7 @@ import shlex
 
 # For conversion from markdown to html
 import recommonmark.parser
+from recommonmark.transform import AutoStructify
 
 # Set paths
 #sys.path.insert(0, os.path.abspath('.'))
@@ -325,6 +326,7 @@ texinfo_documents = [
      'Miscellaneous'),
 ]
 
+
 # Documents to append as an appendix to all manuals.
 #
 # texinfo_appendices = []
@@ -352,3 +354,11 @@ if not on_rtd:
     html_theme_path = [sphinx_rtd_theme.get_html_theme_path()]
 
 # readthedocs.org uses their theme by default, so no need to specify it
+
+
+# At the bottom of conf.py
+def setup(app):
+    app.add_config_value('recommonmark_config', {
+        'enable_auto_doc_ref': True
+    }, True)
+    app.add_transform(AutoStructify)

--- a/tutorial/conf.py
+++ b/tutorial/conf.py
@@ -356,7 +356,7 @@ if not on_rtd:
 # readthedocs.org uses their theme by default, so no need to specify it
 
 
-# At the bottom of conf.py
+# See http://recommonmark.readthedocs.io/en/latest/auto_structify.html
 def setup(app):
     app.add_config_value('recommonmark_config', {
         'enable_auto_doc_ref': True

--- a/tutorial/documents.md
+++ b/tutorial/documents.md
@@ -4,7 +4,7 @@ JupyterLab can be extended in two ways via:
 
 - application plugins (top level): Application plugins extend the
   functionality of JupyterLab itself, and we cover them in the
-  [Plugins](plugins.html) tutorial focuses.
+  [Plugins](plugins.md) tutorial focuses.
 - **document widget extensions (lower level):** Document widget extensions extend
   the functionality of document widgets added to the application, and we cover
   them in this section.

--- a/tutorial/labextensions.md
+++ b/tutorial/labextensions.md
@@ -1,6 +1,6 @@
 # JupyterLab Extensions
 
-A JupyterLab extension provides additional, optional functionality to JupyterLab's built-in capabilities. An extension is a module that provides one or more [plugins](plugins.html) to the JupyterLab application. To streamline third-party development of extensions, this library provides a build script for generating third party extension JavaScript bundles.
+A JupyterLab extension provides additional, optional functionality to JupyterLab's built-in capabilities. An extension is a module that provides one or more [plugins](plugins.md) to the JupyterLab application. To streamline third-party development of extensions, this library provides a build script for generating third party extension JavaScript bundles.
 
 For information on creating a lab extension, please see the documentation for
 the [@jupyterlab/extension-builder](https://github.com/jupyterlab/extension-builder).

